### PR TITLE
Cw update adapter to include zarr outputs

### PR DESCRIPTION
**Pull request by samanehsan**
Opened this PR on behalf of @rexwangcc, who is currently having computer issues 😓 

Please ensure the following when opening a PR:
- [ ] Added or updated tests
- [ ] Updated documentation
- [ ] Applied style guidelines
- [ ] Considered generalizability beyond our own use case

### CODE REVIEW

File: adapter_pipelines/format_map_example.json [hunk 1]
```diff
@@ -1,12 +1,18 @@
...
+  ".theta$":"theta",
+  ".cnt$":"cnt",
+  ".time$":"time",
+  ".model$":"model",
+  ".bai$": "bai",
+  ".tsv$": "tsv",
+  ".mtx$": "mtx",
+  ".h5$": "h5",
+  ".csv$": "csv",
+  ".zarr!": "zarr_format"
```
Comment (samanehsan) [inline adapter_pipelines/format_map_example.json]: Why would this check `.zarr!` and not `.zarray`?

File: pipeline_tools/tests/test_create_analysis_metadata.py [hunk 2]
```diff
@@ -180,10 +180,12 @@ def test_get_workflow_tasks(self, data_file):
 
     def test_get_file_format(self):
         assert cam.get_file_format('asdf', {}) == 'unknown'
-        assert cam.get_file_format('asdf.bam', {'.bam': 'bam'}) == 'bam'
-        assert cam.get_file_format('asdf.txt', {'.bam': 'bam'}) == 'unknown'
-        assert cam.get_file_format('asdf.bam', {'.bam': 'bam', '_metrics': 'metrics'}) == 'bam'
-        assert cam.get_file_format('asdf.foo_metrics', {'.bam': 'bam', '_metrics': 'metrics'}) == 'metrics'
+        assert cam.get_file_format('asdf.bam', {'.bam$': 'bam'}) == 'bam'
+        assert cam.get_file_format('asdf.txt', {'.bam$': 'bam'}) == 'unknown'
```
Comment (samanehsan) [inline pipeline_tools/tests/test_create_analysis_metadata.py]: We may not have this situation with the SS2 outputs, but what if a file matches multiple patterns? E.g. `test_metrics.txt`

File: adapter_pipelines/format_map_example.json (hunk 1 above)
Comment (rexwangcc) [reply adapter_pipelines/format_map_example.json]: `.zarray` is just one of the "`zarr`" files family, in this case, `.zarr!` indeed means `.zarr/` directory before we did the naming trick. So anything contains ".zarr!" was a zarr format file.

File: pipeline_tools/tests/test_create_analysis_metadata.py (hunk 2 above)
Comment (rexwangcc) [reply pipeline_tools/tests/test_create_analysis_metadata.py]: With `$` in the regex, we won't run into that situation. For example, `test_metrics.txt` will only be matched by `.txt$`

**Comment by rexwangcc** [comment]
retest this please

**Comment by rexwangcc** [comment]
retest this please

**Comment by rexwangcc** [comment]
The Mintegration test already passed on: https://job-manager.caas-prod.broadinstitute.org/jobs/8bc95cc8-3b19-4d1b-a247-05e91c9deb8d we can merge this PR with confidence!

**Pull request merged**